src/hooks/useAccounts.tsx

```diff
@@ -188,6 +195,19 @@ const useAccountsContext = () => {
     }
   }, [evmAddress, dydxAddress]);
 
+  const selectedDydxChainId = useSelector(getSelectedDydxChainId);
+  const selectedNetwork = useSelector(getSelectedNetwork);
+  const chainId = Number(ENVIRONMENT_CONFIG_MAP[selectedNetwork].ethereumChainId);
+
+  const signTypedData = getSignTypedData(selectedDydxChainId);
+  const { signTypedDataAsync } = useSignTypedData({
+    ...signTypedData,
+    domain: {
+      ...signTypedData.domain,
+      chainId,
+    },
+  });
```

Comment:
Used in `GenerateKeys.tsx` as well. Could create a new hook for this action.

Reply:
+ 1 on this

Reply:
what would you name the hook?

Reply:
Maybe something like `useSignForWalletDerivation`